src/main/kotlin/org/rust/ide/annotator/RsCargoCheckAnnotator.kt

```diff
@@ -147,18 +147,23 @@ private fun filterMessage(file: PsiFile, document: Document, message: RustcMessa
     val spanFilePath = PathUtil.toSystemIndependentName(span.file_name)
     if (!file.virtualFile.path.endsWith(spanFilePath)) return null
 
+    @Suppress("NAME_SHADOWING")
     fun toOffset(line: Int, column: Int): Int? {
+        val line = line - 1
+        val column = column - 1
```

Comment:
>I am currently working in Elixir

\*green with envy\*

>can introduce nasty bugs when you forget that there is a shadow somewhere in a function.

In my own code, I think I find the opposite is true more often: without shadowing, I often have two variables of the same type in scope, like `foo` and `foo_some_suffix`, and then I use the original `foo` while I should have used `foo_some_suffix`, and even Haskell's type system does not help me. With shadowing, there's only one variable in scope, so you can't use the wrong one =P To me, shadowing is more about Bloch's "minimize visibility", rather then "mutability". 

Would be great to be able to measure and quantify the effects of shadowing though!